packages/mdc-switch/mdc-switch.scss

```diff
@@ -0,0 +1,202 @@
+        position: absolute;
+        top: -14px;
+        left: -14px;
+        width: $mdc-switch-focus-ring-diameter;
+        height: $mdc-switch-focus-ring-diameter;
+        transform: scale(0);
+        transition:
+          mdc-checkbox-transition(transform, $mdc-switch-transition-curve),
+          mdc-checkbox-transition(background-color, $mdc-switch-transition-curve);
+        border-radius: $mdc-switch-focus-ring-diameter / 2;
+        background-color: transparent;
+        content: "";
+        opacity: .2;
+      }
+    }
+  }
+}
+
+.mdc-switch__native-control:focus ~ .mdc-switch__background {
+
+  .mdc-switch__knob {
+    &::before {
+      position: absolute;
+      width: $mdc-switch-focus-ring-diameter;
+      height: $mdc-switch-focus-ring-diameter;
+      transform: scale(1);
+      transition:
+        mdc-checkbox-transition(transform, $mdc-switch-transition-curve),
+        mdc-checkbox-transition(background-color, $mdc-switch-transition-curve);
+      border-radius: $mdc-switch-focus-ring-diameter / 2;
+      background-color: $mdc-switch-unchecked-focus-ring-color;
+
+      @include mdc-theme-dark(".mdc-checkbox") {
+        background-color: $mdc-switch-unchecked-knob-color;
+      }
+
+    }
+  }
+}
+
+.mdc-switch__native-control:checked ~ .mdc-switch__background {
+
+  &::before {
+    @include mdc-theme-prop(background-color, primary);
+    opacity: .5;
+  }
+
+  .mdc-switch__knob {
+    transform: translateX($mdc-switch-track-width - $mdc-switch-knob-diameter);
+    transition:
+      mdc-checkbox-transition(transform, $mdc-switch-transition-curve),
+      mdc-checkbox-transition(background-color, $mdc-switch-transition-curve);
+
+    @include mdc-theme-prop(background-color, primary);
+
+    &::before {
+      @include mdc-theme-prop(background-color, primary);
+
+      @include mdc-theme-dark(".mdc-checkbox") {
+        @include mdc-theme-prop(background-color, primary);
+      }
+
+      opacity: .15;
+    }
+  }
+}
+
+// postcss-bem-linter: end
+
+.mdc-switch--disabled {
+  cursor: initial;
+
+  .mdc-switch__native-control {
+    cursor: initial;
+  }
+
+  .mdc-switch__background {
+    cursor: initial;
+
+    &::before {
+      background-color: $mdc-switch-unchecked-track-color-dark;
+
+      @include mdc-theme-dark(".mdc-checkbox") {
+        background-color: $mdc-switch-unchecked-track-color-dark;
+      }
+
+    }
+
+    .mdc-switch__knob {
+      background-color: $mdc-switch-unchecked-knob-color;
+
+      @include mdc-theme-dark(".mdc-checkbox") {
+        background-color: $mdc-switch-unchecked-knob-color-dark;
+      }
+
+    }
+  }
+
+  .mdc-switch__native-control:focus ~ .mdc-switch__knob::before {
+    transform: scale(1);
+    transition:
+      mdc-checkbox-transition(transform, $mdc-switch-transition-curve),
+      mdc-checkbox-transition(background-color, $mdc-switch-transition-curve);
+    background-color: $mdc-switch-disabled-focus-ring-color;
+    opacity: .1;
+  }
+}
+
+.mdc-switch-label {
```

Comment:
I'd remove this. Things like this are handled by [mdc-form-field](../../tree/master/packages/mdc-form-field). Thus, I'd probably also remove the "mdc-switch-label" class from the demos as well.